MaxText/train.py

```diff
@@ -209,20 +208,20 @@ def loss_fn(params):
   if config.record_internal_nn_metrics:
     record_activation_metrics(metrics, intermediate_outputs, config)
 
-  return new_state, metrics, rng2
+  return new_state, metrics
```

Comment:
You should change `[get_functional_train_with_signature](https://github.com/google/maxtext/blob/65d1d164716be5ab820b1f0d9a2d946297056399/MaxText/maxtext_utils.py#L42C32-L42C32)` to remove the rng output - I think AOT needs this. Maybe we need IF CHANGE THIS CODE THEN CHANGE THAT CODE comments for changing train step signature